src/modules/Game.class.js

```diff
@@ -50,19 +103,130 @@
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.state;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    const initialStateEmpty = this.isInitialStateAllZeros;
+
+    const newInitialState = this.generateInitialState(initialStateEmpty);
+
+    this.resetGamesState(newInitialState);
+
+    return this.initialState;
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.score = 0;
+    this.state = 'idle';
+
+    return this.initialState;
+  }
+
+  // add metodo de inicialização do jogo
+  isInitialStateAllZeros() {
+    return this.initialState.flat().every((value) => value === 0);
+  }
+
+  generateInitialState(isEmpty) {
+    const numberOfCellsToAdd = isEmpty ? 2 : 1;
+
+    return this.addNewCellRandomCell(this.initialState, numberOfCellsToAdd);
+  }
+
+  resetGamesState(newState) {
+    this.initialState = newState;
+    this.score = 0;
+    this.state = 'playing';
+  }
+
+  // add metodo de atualização do tabuleiro quando se move para a esquerda
+  processMoveLeft() {
+    const moveRows = this.initialState.map(this.mergeToTheLeft);
+    const withNewCell = this.addNewCellRandomCell(moveRows, 1);
+
+    return withNewCell;
+  }
+
+  // atualiza os movimentos independente da direção
+  updatedGameState(newState) {
+    this.state = this.loseWinState(newState);
+    this.score += this.calculateScore(this.initialState, newState);
+    this.initialState = newState;
+  }
+
+  // add metodo de atualização do tabuleiro quando se move para a direta
+  processMoveRight() {
+    const moveRows = this.initialState(this.mergeToTheRight);
+    const withNewCell = this.addNewCellRandomCell(moveRows, 1);
+
+    return withNewCell;
+  }
+
+  // add metodo de atualização do tabuleiro quando se move para cima
+  processMoveUp() {
+    const moveRows = this.initialState(this.mergeToTheUp);
+    const withNewCell = this.addNewCellRandomCell(moveRows, 1);
+
+    return withNewCell;
+  }
+
+  // add metodo de atualização do tabuleiro quando se move para baixo
+  processMoveDown() {
+    const moveRows = this.initialState(this.mergeToTheDown);
+    const withNewCell = this.addNewCellRandomCell(moveRows, 1);
+
+    return withNewCell;
+  }
+
+  // add novas celulas ao tabuleiro
+  addNewCellRandomCell(initialState, qtdCellAdd) {
+    const matrixInRow = initialState.flat();
+
+    const indexEvaluation = matrixInRow.forEach((value, index) => {
+      if (value === 0 && qtdCellAdd > 0) {
+        matrixInRow[index] = 1;
+      } else {
+        matrixInRow[index] = null;
+      }
+    });
+
+    if (indexEvaluation.length === 0) {
+      return initialState;
+    }
+  }
+
+  // verifica se perdeu ou ganhou
+  loseWinState() {}
+
+  // verifca a pontuação
+  calculateScore() {}
+
+  // mescla para a esquerda
+  mergeToTheLeft() {}
```

Comment:
The method `mergeToTheLeft` is not implemented. It should handle merging tiles to the left.